crates/rattler_virtual_packages/src/cuda.rs

```diff
@@ -1,29 +1,321 @@
+
+    let cu_device_get_attribute: Symbol<
+        '_,
+        unsafe extern "C" fn(*mut c_int, c_int, c_int) -> c_ulong,
+    > = unsafe { cuda_library.get(b"cuDeviceGetAttribute\0") }.ok()?;
+
+    let cu_device_get_name: Symbol<'_, unsafe extern "C" fn(*mut u8, c_int, c_int) -> c_ulong> =
+        unsafe { cuda_library.get(b"cuDeviceGetName\0") }.ok()?;
+
+    // Get the number of CUDA devices
+    let mut device_count = MaybeUninit::uninit();
+    if unsafe { cu_device_get_count(device_count.as_mut_ptr()) } != 0 {
+        return None;
+    }
+    let device_count = unsafe { device_count.assume_init() };
+
+    // No devices found
+    if device_count == 0 {
+        return None;
+    }
+
+    // Iterate through all devices to find the minimum compute capability
+    let mut min_arch: Option<CudaArchInfo> = None;
+
+    for device_idx in 0..device_count {
+        // Get device handle
+        let mut device = MaybeUninit::uninit();
+        if unsafe { cu_device_get(device.as_mut_ptr(), device_idx) } != 0 {
+            continue;
+        }
+        let device = unsafe { device.assume_init() };
+
+        // Get compute capability major version
+        let mut cc_major = MaybeUninit::uninit();
+        if unsafe {
+            cu_device_get_attribute(
+                cc_major.as_mut_ptr(),
+                CU_DEVICE_ATTRIBUTE_COMPUTE_CAPABILITY_MAJOR,
+                device,
+            )
+        } != 0
+        {
+            continue;
+        }
+        let cc_major = unsafe { cc_major.assume_init() } as u32;
+
+        // Get compute capability minor version
+        let mut cc_minor = MaybeUninit::uninit();
+        if unsafe {
+            cu_device_get_attribute(
+                cc_minor.as_mut_ptr(),
+                CU_DEVICE_ATTRIBUTE_COMPUTE_CAPABILITY_MINOR,
+                device,
+            )
+        } != 0
+        {
+            continue;
+        }
+        let cc_minor = unsafe { cc_minor.assume_init() } as u32;
+
+        // Check if this is the minimum compute capability so far
+        let is_new_minimum = min_arch.as_ref().is_none_or(|min| {
+            cc_major < min.major || (cc_major == min.major && cc_minor < min.minor)
+        });
+
+        if is_new_minimum {
+            // Get device name
+            let mut name_buffer = [0u8; MAX_DEVICE_NAME_LEN];
+            if unsafe {
+                cu_device_get_name(
+                    name_buffer.as_mut_ptr(),
+                    MAX_DEVICE_NAME_LEN as c_int,
+                    device,
+                )
+            } == 0
+            {
+                // Convert C string to Rust string
+                if let Ok(cstr) = CStr::from_bytes_until_nul(&name_buffer) {
+                    if let Ok(device_name) = cstr.to_str() {
+                        min_arch = Some(CudaArchInfo {
+                            major: cc_major,
+                            minor: cc_minor,
+                            device_name: device_name.to_string(),
```

Comment:
https://github.com/conda-incubator/nvidia-virtual-packages/pull/5

AFAICT, this MR is missing some post-processing that I do to the device string. I strip the substring "NVIDIA" from the device name to make it shorter. Also, today I started some work on making the plugin CEP-26 compliant. This CEP limits the length of a build string and states which chars are allowed.